File: fizzbuzz.php
<!DOCTYPE html>
<html>
<body>
<?php
$dict = array(
  3 => "Fizz",  
  5 => "Buzz", 
);
$output = "";
for ($i = 1; $i <= 100; $i++) {
  $output = "";
  foreach( $dict as $key => $value ){
    if ($i % $key == 0) {
      $output .= $value;
    }
  }
  if ($output == "") {
    $output = $i;
  }
  echo "<div>$output</div>";
}
?>
</body>
</html>

File: fizzbuzz.java
import java.util.Map; 
import java.util.TreeMap;

public class FizzBuzz {

     public static void main(String []args) {
        String output;
        TreeMap<Integer, String> dict = new TreeMap<Integer, String>();
        dict.put(3, "Fizz"); 
        dict.put(5, "Buzz");
        for (int i = 1; i <= 100; i++) {
            output = "";
			// The set's iterator returns the entries in ascending key order: https://docs.oracle.com/javase/7/docs/api/java/util/TreeMap.html#entrySet()
            for (Map.Entry<Integer, String> entry : dict.entrySet()) {
                if (i % entry.getKey() == 0) { output += entry.getValue(); }
		    }
		    if (output.equals("")) { output = String.valueOf(i); }
		    System.out.println(output);
        }
     }
}